src/ts/services/keyring-vault-proxy.ts

```diff
@@ -0,0 +1,174 @@
+import { browser } from 'webextension-polyfill-ts'
+import { KeyringPair$Json } from '@polkadot/keyring/types'
+
+export function isWalletLocked (): Promise<boolean> {
+  const port = browser.runtime.connect(undefined, { name: '__SPECKLE__' })
```

Comment:
Is it possible to connect once and then the connection can be shared without disconnecting?